### PR TITLE
Geolocation temp

### PR DESCRIPTION
**Pull request by nickdesaulniers**
review? @mykmelez

This may not apply cleanly due to all the merge commits, but if not it's really just the last commit that is important and can be cleanly applied or manually fudged to work.  The important thing is to start the review process.

### CODE REVIEW

File: addon/package.json [hunk 1]
```diff
@@ -8,6 +8,7 @@
   "contributors": [
     "Anant Narayanan (https://github.com/anantn)",
     "Harald Kirschner (https://github.com/digitarald)",
+    "Kan-Ru Chen (陳侃如) (https://github.com/kanru)",
     "Luca Greco (https://github.com/rpl)",
     "Matthew Riley MacPherson (https://github.com/tofumatt)",
```
Comment (mykmelez) [inline addon/package.json:13]: Don't forget to add yourself! We do it alphabetically by first name, so you would be between Matthew and Victor:

```
    "Nick Desaulniers (https://github.com/nickdesaulniers)",
```

File: prosthesis/components/FakeGeolocationProvider.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+function FakeGeoCoordsObject(lat, lon, acc, alt, altacc) {
+  this.latitude = lat;
+  this.longitude = lon;
+  this.accuracy = acc;
+  this.altitude = alt;
+  this.altitudeAccuracy = altacc;
+}
+
+FakeGeoCoordsObject.prototype = {
+
```
Comment (mykmelez) [inline prosthesis/components/FakeGeolocationProvider.js:20]: Nit: remove this unconventional line of whitespace.

File: prosthesis/components/FakeGeolocationProvider.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+                                           Ci.nsIFakeListener,
+                                           Ci.nsITimerCallback]),
+  startup:  function() {
+    Services.obs.notifyObservers(null, "r2d2b2g-geolocation-request", null);
+    let self = this;
+    Services.obs.addObserver(function (message) {
+      self.currentLoc = new FakeGeoPositionObject(message.wrappedJSObject.lat,
+        message.wrappedJSObject.lon);
+      self.callback.update(self.currentLoc);
+    }, "r2d2b2g-geolocation-response", false);
```
Comment (mykmelez) [inline prosthesis/components/FakeGeolocationProvider.js:60]: Nit: `Function.bind()` is the new hotness for accessing an outer function's _this_ object from an inner function:

```
Services.obs.addObserver((function (message) {
  this.currentLoc = new FakeGeoPositionObject(message.wrappedJSObject.lat,
    message.wrappedJSObject.lon);
  this.callback.update(this.currentLoc);
}).bind(this), "r2d2b2g-geolocation-response", false);
```

File: prosthesis/components/FakeGeolocationProvider.js [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+  classID:          Components.ID("{a93105f2-8169-4790-a455-4701ce867aa8}"),
+  QueryInterface:   XPCOMUtils.generateQI([Ci.nsIGeolocationProvider,
+                                           Ci.nsIFakeListener,
+                                           Ci.nsITimerCallback]),
+  startup:  function() {
+    Services.obs.notifyObservers(null, "r2d2b2g-geolocation-request", null);
+    let self = this;
+    Services.obs.addObserver(function (message) {
+      self.currentLoc = new FakeGeoPositionObject(message.wrappedJSObject.lat,
+        message.wrappedJSObject.lon);
```
Comment (mykmelez) [inline prosthesis/components/FakeGeolocationProvider.js:58]: Nit: Firefox code conventionally aligns the arguments in a multi-line function call like this, i.e.:

```
  this.currentLoc = new FakeGeoPositionObject(message.wrappedJSObject.lat,
                                              message.wrappedJSObject.lon);
```

File: addon/lib/remote-simulator-client.js [hunk 5]
```diff
@@ -199,6 +204,25 @@ const RemoteSimulatorClient = Class({
       emit(this, "clientClosed", {client: client});
     }).bind(this));
 
+    client.addListener("geolocationRequest", function () {
+      let geolocation = Cc["@mozilla.org/geolocation;1"].getService(
+        Ci.nsIDOMGeoGeolocation);
```
Comment (mykmelez) [inline addon/lib/remote-simulator-client.js:209]: Nit: Firefox code conventionally formats such statements as:

```
let geolocation = Cc["@mozilla.org/geolocation;1"].
                  getService(Ci.nsIDOMGeoGeolocation);
```

File: addon/lib/remote-simulator-client.js [hunk 6]
```diff
@@ -199,6 +204,25 @@ const RemoteSimulatorClient = Class({
...
+      geolocation.getCurrentPosition(function success (position) {
```
Comment (mykmelez) [inline addon/lib/remote-simulator-client.js:210]: Nit: remove the space between the function name and its parameter list (here and on the _error_ function below).

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+10. Run it! `cfx run --templatedir template/ --profiledir ../../arbitraryProfileName/`
+
+Note: to close it use commmand quit in osx, control-c from the command line will not save any newly installed apps.
+
+# Contributing #
+
+Pick up an issue from the [issue tracker](https://github.com/mozilla/r2d2b2g/issues?state=open) and checkout a new branch that describes the issue.
+ex. `git checkout -b fixThatBrokenThingYouKnow`
+
+squashed commits preferred, then issue a pull request.
```
Comment (mykmelez) [inline CONTRIBUTING.md:21]: I merged your new CONTRIBUTING.md file from your other pull request! So if you fetch upstream and merge the latest upstream/master into this branch, you'll get a conflict in this file. But simply resolve it in favor of master's version of the file, and all will be good.

File: prosthesis/components/FakeGeolocationProvider.js (hunk 3 above)
Comment (nickdesaulniers) [inline prosthesis/components/FakeGeolocationProvider.js:60]: That's a cool pattern; I never thought about using bind on an anonymous function literal!

**Comment by nickdesaulniers** [comment]
Thanks for the review @mykmelez , new pull request inbound

**Pull request closed without merging**